File: first_dup.rb
def firstDuplicate(a)
    temp = []
    duplicates = []
    a.each do |n|
       duplicates << n  if temp.bsearch {|x| x == n} 
       temp << n unless temp.bsearch {|x| x == n} 
    end
    return -1 if duplicates.count == 0
    return duplicates.first unless duplicates.count == 0
end
a=[]
limit = 100_000
limit.times { |n| a << rand(1..limit) }
puts firstDuplicate a

File: FirstDup.cs
using System;
using System.Linq;
using System.Collections.Generic;

namespace CodeFight
{
    public class Programm 
    {
        public static void Main(string[] args)
        {
            int[] a = {2, 3, 3, 1, 5, 2};
            Console.WriteLine(firstDuplicate(a).ToString());
        }

        public static int firstDuplicate(int[] a) 
        {
            List<int> temp = new List<int>();
            List<int> duplicates = new List<int>();
            foreach (int n in a)
            {
                if(temp.Contains(n))
                    duplicates.Add(n);
                else
                    temp.Add(n);
            }
            if(duplicates.Count == 0)
                return -1;
            else
                return duplicates.First();
        }
    }

    
}
